pkg/apiserver/debugapi/service.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugapi
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joomcode/errorx"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/user"
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/utils"
+)
+
+var (
+	ErrNS                = errorx.NewNamespace("error.api.debugapi")
+	ErrComponentClient   = ErrNS.NewType("invalid_component_client")
+	ErrEndpointConfig    = ErrNS.NewType("invalid_endpoint_config")
+	ErrInvalidStatusPort = ErrNS.NewType("invalid_status_port")
+)
+
+func registerRouter(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
+	endpoint := r.Group("/debugapi")
+	endpoint.Use(auth.MWAuthRequired())
+
+	endpoint.POST("/endpoint", s.RequestEndpoint)
```

Comment:
It might be good to introduce a verb to the POST interface, like `request_endpoint`.